cdap-api-common/src/main/test/java/co/cask/cdap/api/data/schema/SchemaWalkerTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.data.schema;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Tests the schema walker.
+ */
+public class SchemaWalkerTest {
+
+  @Test
+  public void testSchemaWalker() {
+    Schema bytesArraySchema = Schema.arrayOf(Schema.of(Schema.Type.BYTES));
```

Comment:
Should we add a test for invalid schema to trigger default case of the switch?

Reply:
hmmm - how? The default case is for the future. If we ever add another type, then the schema walker must fail instead of silently ignoring it. But right now it is not possible to create a schema with a different type.